test/e2e/lazy_activation_policy_test.go

```diff
@@ -17,57 +17,241 @@ limitations under the License.
-
-			policy = testhelper.NewLazyPropagationPolicy(policyNamespace, policyName, []policyv1alpha1.ResourceSelector{
-				{
-					APIVersion: deployment.APIVersion,
-					Kind:       deployment.Kind,
-					Name:       deployment.Name,
-				}}, policyv1alpha1.Placement{
-				ClusterAffinity: &policyv1alpha1.ClusterAffinity{
-					ClusterNames: []string{targetMember},
-				},
-			})
+	var namespace string
+	var deploymentName, configMapName, policyName string
+	var originalCluster, modifiedClutser string
+	var deployment *appsv1.Deployment
+	var configMap *corev1.ConfigMap
+	var policy *policyv1alpha1.PropagationPolicy
+
+	ginkgo.BeforeEach(func() {
+		namespace = testNamespace
+		deploymentName = deploymentNamePrefix + rand.String(RandomStrLength)
+		configMapName = deploymentName
+		policyName = deploymentName
+		originalCluster = framework.ClusterNames()[0]
+		modifiedClutser = framework.ClusterNames()[1]
+
+		deployment = testhelper.NewDeployment(namespace, deploymentName)
+		configMap = testhelper.NewConfigMap(namespace, configMapName, map[string]string{"test": "test"})
+		policy = testhelper.NewLazyPropagationPolicy(namespace, policyName, []policyv1alpha1.ResourceSelector{
+			{
+				APIVersion: deployment.APIVersion,
+				Kind:       deployment.Kind,
+				Name:       deploymentName,
+			}}, policyv1alpha1.Placement{
+			ClusterAffinity: &policyv1alpha1.ClusterAffinity{
+				ClusterNames: []string{originalCluster},
+			},
 		})
+	})
 
-		ginkgo.BeforeEach(func() {
+	ginkgo.Context("1. Policy created before resource", func() {
+		ginkgo.JustBeforeEach(func() {
 			framework.CreatePropagationPolicy(karmadaClient, policy)
+			waiPropagatePolicyExistInKarmada(namespace, policyName)
+			framework.CreateDeployment(kubeClient, deployment)
+
+			ginkgo.DeferCleanup(func() {
+				framework.RemovePropagationPolicyIfExist(karmadaClient, namespace, policyName)
+				framework.RemoveDeployment(kubeClient, namespace, deploymentName)
+				framework.WaitDeploymentDisappearOnCluster(originalCluster, namespace, deploymentName)
+			})
+		})
+
+		// Simple Case 1 (Policy created before resource)
+		// refer: https://github.com/karmada-io/karmada/blob/master/docs/proposals/scheduling/activation-preference/lazy-activation-preference.md#simple-case-1-policy-created-before-resource
+		ginkgo.It("Simple Case 1 (Policy created before resource)", func() {
+			ginkgo.By("step 1: deployment propagate success when policy created before it", func() {
+				waitDeploymentPresentOnCluster(originalCluster, namespace, deploymentName)
+			})
+
+			ginkgo.By("step 2: after policy deleted, deployment still keep previous propagation states", func() {
+				framework.RemovePropagationPolicy(karmadaClient, namespace, policyName)
+				// wait to distinguish whether the state will not change or have no time to change
+				time.Sleep(pollInterval)
+				waitDeploymentPresentOnCluster(originalCluster, namespace, deploymentName)
```

Comment:
since our real purpose is to **wait** to distinguish whether the state will not change or have no time to change, `waitDeploymentPresentOnCluster` is acceptable to a certain extent, maybe not the clearest, but judging whether "have no time to change" do really need a **wait**.